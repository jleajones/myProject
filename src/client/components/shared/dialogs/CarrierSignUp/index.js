import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import CloseIcon from '@material-ui/icons/Close';
import IconButton from '@material-ui/core/IconButton';
import Toolbar from '@material-ui/core/Toolbar';
import AppBar from '@material-ui/core/AppBar';
import makeStyles from '@material-ui/core/styles/makeStyles';

import Dialog from '../Dialog';

const useStyles = makeStyles(theme => ({
  title: {
    marginLeft: theme.spacing(2),
    flex: 1
  }
}));

const CarrierSignUp = ({ handleClose, open }) => {
  const classes = useStyles();
  const title = 'Ship With Us';
  return (
    <Dialog
      onClose={handleClose}
      label="ship-with-us-dialog-title"
      title={title}
      open={open}
      fullScreen
    >
      <AppBar>
        <Toolbar>
          <IconButton
            edge="start"
            color="inherit"
            onClick={handleClose}
            aria-label="close"
          >
            <CloseIcon />
          </IconButton>
          <Typography variant="h6" className={classes.title}>
            {title}
          </Typography>
          <Button color="inherit" onClick={handleClose}>
            save
          </Button>
        </Toolbar>
      </AppBar>
    </Dialog>
  );
};

CarrierSignUp.defaultProps = {
  open: false
};

CarrierSignUp.propTypes = {
  handleClose: PropTypes.func.isRequired,
  open: PropTypes.bool
};

export default CarrierSignUp;
