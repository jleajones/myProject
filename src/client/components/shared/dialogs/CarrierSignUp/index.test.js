import React from 'react';
import { shallow } from 'enzyme';

import CarrierSignUp from './index';

describe('<CarrierSignUp />', () => {
  test('Renders component without crashing', () => {
    const handleCloseDialog = jest.fn();
    const open = false;
    const component = shallow(
      <CarrierSignUp handleClose={handleCloseDialog} open={open} />
    );

    expect(component).toMatchSnapshot();
  });
});
