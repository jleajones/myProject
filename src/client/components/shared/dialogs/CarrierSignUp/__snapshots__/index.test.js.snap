// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarrierSignUp /> Renders component without crashing 1`] = `
<Dialog
  fullScreen={true}
  label="ship-with-us-dialog-title"
  onClose={[MockFunction]}
  open={false}
  title="Ship With Us"
>
  <WithStyles(ForwardRef(AppBar))>
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="close"
        color="inherit"
        edge="start"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-1"
        variant="h6"
      >
        Ship With Us
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button))
        color="inherit"
        onClick={[MockFunction]}
      >
        save
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</Dialog>
`;
